### PR TITLE
Add list projects

### PR DESCRIPTION
**Pull request by jtsaito**
Cleaning up the clients class is out of scope.

### CODE REVIEW

File: rollbar/testdata/fixtures/projects/list.json [hunk 1]
```diff
@@ -0,0 +1,734 @@
...
+          "email": {},
+          "flowdock": {}
+        },
+        "time_format": "24h",
+        "cleanup": {
+          "production.autoresolve_days": "10"
+        },
+        "timezone": "Europe/Berlin",
+        "grouping": {
+          "global": [
```
Comment (jansiwy) [inline rollbar/testdata/fixtures/projects/list.json]: please reduce the fixture to the relevant parts: e.g. the grouping JSON is custom data and might be just empty in a new simple Rollbar project.

File: rollbar/projects.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package rollbar
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// ListProjectsResponse represents the list projects response
+type ListProjectsResponse struct {
```
Comment (jansiwy) [inline rollbar/projects.go]: this is just a struct for parsing the JSON. IMO it should be private.

File: rollbar/projects.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		ID           int    `json:"id"`
+		DateCreated  int    `json:"date_created"`
+		DateModified int    `json:"date_modified"`
+		Name         string `json:"name"`
+	}
+}
+
+// ListProjects lists the projects for this API Key
+// https://docs.rollbar.com/reference#list-all-projects
+func (c *Client) ListProjects() (*ListProjectsResponse, error) {
```
Comment (jansiwy) [inline rollbar/projects.go]: this seems to be too low level as return type, i'd prefer something like `[]*Project` (like making `ListProjectsResponse` private and using it only internally)

In the new struct you might also drop attributes which are not very useful.

File: rollbar/projects.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+// listProjectsResponse represents the list projects response
+type listProjectsResponse struct {
+	Error  int `json:"err"`
+	Result []Project
+}
+
+// ListProjects lists the projects for this API Key
+// https://docs.rollbar.com/reference#list-all-projects
+func (c *Client) ListProjects() ([]Project, error) {
```
Comment (jansiwy) [inline rollbar/projects.go]: why not `[]*Project`?

**Review by jansiwy — APPROVED** [review]
beyond that: +1

File: rollbar/projects.go (hunk 4 above)
Comment (jtsaito) [reply rollbar/projects.go]: done also

**Pull request merged**